backend/kernelCI_app/views/treeDetailsView.py

```diff
@@ -93,6 +93,32 @@ def create_summary(self, builds_dict):
             "architectures": arch_summ,
         }
 
+    def _get_filters(self, query, filter_params, build_fields, checkout_fields):
+        grouped_filters = {}
+
+        for f in filter_params.filters:
+            field = f["field"]
+            value = f['value']
+            if field not in grouped_filters:
+                grouped_filters[field] = f
+                grouped_filters[field]['value'] = value
+            elif type(grouped_filters[field]['value']) is str:
+                grouped_filters[field]['value'] = [grouped_filters[field]['value'], value]
+            else:
+                grouped_filters[field]['value'].append(value)
```

Comment:
This logic is the same used in `_get_grouped_filters` function in `treeCommitsHistory` file. Better to separate into a function.

Reply:
I moved this to `FilterParams` class